lib/SILOptimizer/Utils/Local.cpp

```diff
@@ -1677,6 +1678,25 @@ optimizeBridgedSwiftToObjCCast(SILInstruction *Inst,
   SILType SubstFnTy = SILFnTy.substGenericArgs(M, SubMap);
   SILFunctionConventions substConv(SubstFnTy.castTo<SILFunctionType>(), M);
 
+  if (Dest) {
+    // check that we can go through with the optimization
+    auto ConvTy = substConv.getSILResultType().getObjectType();
+    auto DestTy = Dest->getType().getObjectType();
```

Comment:
It would be nice if you could add a comment here explaining that DestTy is the type which we want to convert to and ConvTy is the return type of the _bridgeToObjectiveCImpl().
And that both types are ObjC class types. Maybe add an assert.